### PR TITLE
[MySql][Bug] The MySqlParser.g4 parser has a priority problem

### PR DESCRIPTION
**Pull request by caicancai**
The MySqlParser.g4 parser has a priority problem, symbols such as */% should not have the same priority as +- symbols

### CODE REVIEW

**Comment by caicancai** [comment]
cc @KvanTTT

**Comment by caicancai** [comment]
> There's no additional test to verify the parse tree for expressions has been fixed.

I have used antlr4 plugin of idea to verify it, but I don't know if it is accurate

**Comment by caicancai** [comment]
After restoration：
fixed
Before:
屏幕截图 2023-10-31 172227

**Comment by caicancai** [comment]
cc @kaby76

**Comment by kaby76** [comment]
Are you sure that it works? Shouldn't "3+3*2" be the same as "3+(3*2)", not "(3+3)*2" because multiplication has higher precedence than addition?

**Comment by caicancai** [comment]
From the tree structure I am sure that my modification is valid, 3+3*2 was 6*2=12, but in fact 3+6 = 9

**Comment by caicancai** [comment]
Maybe I don't have a good way to verify that right now

**Comment by kaby76** [comment]
Sorry, I didn't set the branch correctly on my clone, and you gave two pictures in which I picked the wrong picture to analyze.

You're going to need to add a test, something like this:

examples/expressions.sql
```
select 1+2*3-4;
```

examples/expressions.sql.tree
```
(root (sqlStatements (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 1))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 2))) (mathOperator *) (expressionAtom (constant (decimalLiteral 3))))) (mathOperator2 -) (expressionAtom (constant (decimalLiteral 4)))))))) fromClause)))) (emptyStatement_ ;)) <EOF>)
```

**Comment by caicancai** [comment]
Hello, thank you for your reply and comments, examples/expressions. SQL. The tree is how to display, I need to write this part, or say only need to supplement the test

**Comment by kaby76** [comment]
Great! But please check in the expressions.sql.tree file so regressions are detected if someone modifies the grammar and breaks it unexpectedly in the future.

File: sql/mysql/Positive-Technologies/MySqlParser.g4 [hunk 1]
```diff
@@ -2668,7 +2669,11 @@ bitOperator
     ;
 
 mathOperator
-    : '*' | '/' | '%' | DIV | MOD | '+' | '-'
+    : '*' | '/' | '%' | DIV | MOD
+    ;
+
+mathOperator2
+    : '+' | '-'
```
Comment (KvanTTT) [inline sql/mysql/Positive-Technologies/MySqlParser.g4:2676]: I suggest renaming `mathOperator` to `multOperator` (short form of `multiplicative`), `matchOperator2` to `addOperator` (short form of `additive`)

Comment (caicancai) [reply sql/mysql/Positive-Technologies/MySqlParser.g4]: done

**Comment by caicancai** [comment]
> Great! But please check in the expressions.sql.tree file so regressions are detected if someone modifies the grammar and breaks it unexpectedly in the future.

Thank you very much for your patient reply and answer. I have learned a lot

**Comment by caicancai** [comment]
cc @KvanTTT @kaby76

**Comment by kaby76** [comment]
> cc @KvanTTT @kaby76

Not done yet. Please add the file sql/mysql/examples/expressions.sql.tree:
```
(root (sqlStatements (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 1))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 2))) (mathOperator *) (expressionAtom (constant (decimalLiteral 3))))) (mathOperator2 -) (expressionAtom (constant (decimalLiteral 4)))))))) fromClause)))) ; (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 1))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 9))) (mathOperator /) (expressionAtom (constant (decimalLiteral 3))))) (mathOperator2 -) (expressionAtom (constant (decimalLiteral 2)))))))) fromClause)))) ; (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 2))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 9))) (mathOperator %) (expressionAtom (constant (decimalLiteral 2))))) (mathOperator2 -) (expressionAtom (fullColumnName (uid (simpleId 1；))))))))) fromClause))))) <EOF>)
```

**Comment by caicancai** [comment]
> > cc @KvanTTT @kaby76
> 
> Not done yet. Please add the file sql/mysql/examples/expressions.sql.tree:
> 
> ```
> (root (sqlStatements (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 1))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 2))) (mathOperator *) (expressionAtom (constant (decimalLiteral 3))))) (mathOperator2 -) (expressionAtom (constant (decimalLiteral 4)))))))) fromClause)))) ; (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 1))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 9))) (mathOperator /) (expressionAtom (constant (decimalLiteral 3))))) (mathOperator2 -) (expressionAtom (constant (decimalLiteral 2)))))))) fromClause)))) ; (sqlStatement (dmlStatement (selectStatement (querySpecification select (selectElements (selectElement (expression (predicate (expressionAtom (expressionAtom (expressionAtom (constant (decimalLiteral 2))) (mathOperator2 +) (expressionAtom (expressionAtom (constant (decimalLiteral 9))) (mathOperator %) (expressionAtom (constant (decimalLiteral 2))))) (mathOperator2 -) (expressionAtom (fullColumnName (uid (simpleId 1；))))))))) fromClause))))) <EOF>)
> ```

I'm very sorry for the late reply, is it only necessary to add this one, if you need to add other tests to the.tree file, I would like to ask how the.tree file is generated

**Comment by caicancai** [comment]
mvn clean testtree.toStringTree()？

**Comment by kaby76** [comment]
> mvn clean testtree.toStringTree()？

Just `mvn clean test`. See the instructions on testing [here](https://github.com/antlr/grammars-v4/wiki#how-do-i-test-the-grammars). The tree produced is from [toStringTree()](https://www.antlr.org/api/Java/org/antlr/v4/runtime/tree/Trees.html#toStringTree(org.antlr.v4.runtime.tree.Tree)). But, you don't need to write a driver to produce that parenthesized string representation. Let the tester produce it.

**Comment by caicancai** [comment]
> > mvn clean testtree.toStringTree()？
> 
> Just . See the instructions on testing [here](https://github.com/antlr/grammars-v4/wiki#how-do-i-test-the-grammars). The tree produced is from [toStringTree()](https://www.antlr.org/api/Java/org/antlr/v4/runtime/tree/Trees.html#toStringTree(org.antlr.v4.runtime.tree.Tree)). But, you don't need to write a driver to produce that parenthesized string representation. Let the tester produce it.`mvn clean test`

Thank you

**Comment by caicancai** [comment]
cc @KvanTTT

**Comment by kaby76** [comment]
Looks good. Thanks!

**Comment by caicancai** [comment]
Good to merge?

**Pull request merged**